create_shards.c

```diff
@@ -256,6 +253,15 @@ master_create_worker_shards(PG_FUNCTION_ARGS)
 			shardMaxHashToken = INT_MAX;
 		}
 
+		/* insert the shard metadata row along with its min/max values */
+		minHashTokenText = IntegerToText(shardMinHashToken);
+		maxHashTokenText = IntegerToText(shardMaxHashToken);
+		shardId = CreateShardRow(distributedTableId, shardStorageType, minHashTokenText,
```

Comment:
Discussed at length over the past few weeks. Sticking with this new approach. (done)